From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 1 Jan 2024 17:19:30 +0100
Subject: [PATCH] Revert "d3d12: Only destroy the winsys during screen
 destruction, not reset"

This reverts commit 684d11421c30c0f8230ccbcc8dcc94a457eb5f44.

For: https://gitlab.archlinux.org/archlinux/packaging/packages/mesa/-/issues/5
---
 .pick_status.json                          | 2 +-
 src/gallium/drivers/d3d12/d3d12_screen.cpp | 8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/.pick_status.json b/.pick_status.json
index 0c787e915471..f6b0909e7d92 100644
--- a/.pick_status.json
+++ b/.pick_status.json
@@ -2014,7 +2014,7 @@
         "description": "d3d12: Only destroy the winsys during screen destruction, not reset",
         "nominated": true,
         "nomination_type": 1,
-        "resolution": 1,
+        "resolution": 0,
         "main_sha": null,
         "because_sha": "81c8e89ff8e846839fffc1d103b2080bef5c1b5c",
         "notes": null
diff --git a/src/gallium/drivers/d3d12/d3d12_screen.cpp b/src/gallium/drivers/d3d12/d3d12_screen.cpp
index 9b0233fb9d92..ee74cf2b064f 100644
--- a/src/gallium/drivers/d3d12/d3d12_screen.cpp
+++ b/src/gallium/drivers/d3d12/d3d12_screen.cpp
@@ -735,15 +735,15 @@ d3d12_deinit_screen(struct d3d12_screen *screen)
       screen->dev->Release();
       screen->dev = nullptr;
    }
+   if (screen->winsys) {
+      screen->winsys->destroy(screen->winsys);
+      screen->winsys = nullptr;
+   }
 }
 
 void
 d3d12_destroy_screen(struct d3d12_screen *screen)
 {
-   if (screen->winsys) {
-      screen->winsys->destroy(screen->winsys);
-      screen->winsys = nullptr;
-   }
    slab_destroy_parent(&screen->transfer_pool);
    mtx_destroy(&screen->submit_mutex);
    mtx_destroy(&screen->descriptor_pool_mutex);
